src/job_worker/Jobs.jl

```diff
@@ -167,46 +194,207 @@ end
+    @info "Done setting up regional assessment data"
+
+    @info "Performing regional assessment (dependency of site assessment)"
+
+    # Pull out these parameters in the format previously expected 
+    reg = input.region
+    rtype = input.reef_type
+
+    # This is the format expected by prior methods TODO improve the separation
+    # of concerns between query string parameters and function inputs - the API
+    # routing concerns should not be exposed to the application layer
+    qp::Dict{String,String} = Dict{String,String}(
+        "Depth" => "$(input.depth_min):$(input.depth_max)",
+        "Slope" => "$(input.slope_min):$(input.slope_max)",
+        "Rugosity" => "$(input.rugosity_min):$(input.rugosity_max)",
+        "SuitabilityThreshold" => "$(input.threshold)",
+        "xdist" => "$(input.x_dist)",
+        "ydist" => "$(input.y_dist)"
+    )
+
+    @debug "Ascertaining file name"
+    cache_path = _cache_location(config)
+
+    # Use only the criteria relevant to regional assessment
+    regional_assess_criteria = extract_criteria(qp, suitability_criteria())
+    # NOTE: This is where we could add additional hash params to invalidate
+    # cache
+    job_id = create_job_id(regional_assess_criteria)
+
+    assessed_fn = joinpath(cache_path, "$(job_id)_$(reg)_suitable.tiff")
+    @debug "File name: $(assessed_fn)"
+
+    if !isfile(assessed_fn)
+        @debug "File system cache was not hit for this task"
+        @debug "Assessing region $(reg)"
+        assessed = assess_region(reg_assess_data, reg, qp, rtype)
+
+        @debug "Writing to $(assessed_fn)"
+        _write_tiff(assessed_fn, assessed)
+    else
+        @info "Cache hit - skipping regional assessment process!"
+    end
+
+    @debug "Pulling out raster"
+    assessed = Raster(assessed_fn; missingval=0, lazy=true)
+
+    # Extract criteria and assessment
+    pixel_criteria = extract_criteria(qp, search_criteria())
+    deploy_site_criteria = extract_criteria(qp, site_criteria())
+
+    @debug "Performing site assessment"
+    best_sites = filter_sites(
+        assess_sites(
+            reg_assess_data, reg, rtype, pixel_criteria, deploy_site_criteria,
+            assessed
+        )
+    )
+
+    # Specifically clear from memory to invoke garbage collector
+    assessed = nothing
+
+    @debug "Writing to temporary file"
+    geojson_name = "$(tempname()).geojson"
+    @debug "File name $(geojson_name)"
+
+    if nrow(best_sites) == 0
+        open(geojson_name, "w") do f
+            JSON.print(f, nothing)
+        end
+    else
+        output_geojson(geojson_name, best_sites)
+    end
+
+    # Now upload this to s3 
+    client = S3StorageClient(; region=context.aws_region)
+
+    # Output file names
+    output_file_name_rel = "suitable.geojson"
+    full_s3_target = "$(context.storage_uri)/$(output_file_name_rel)"
+    @debug "File paths:" relative = output_file_name_rel absolute = full_s3_target
+
+    upload_file(client, geojson_name, full_s3_target)
```

Comment:
I think okay for now but I would prefer abstracting this level of concern away.

Could create storage interfaces for local/S3/whatever so use here becomes:

```julia
client = SomeStorageClient()
write(client, path, data)
```

Could be overcooking it